include/mo.php

```diff
@@ -73,5 +100,60 @@ public function import_from_db( $lang ) {
 	 */
 	public function delete_entry( $string ) {
 		unset( $this->entries[ $string ] );
+
+		$language = $this->get_header( 'Language' );
+		if ( ! empty( $language ) ) {
+			self::$cache->set( $language, $this->entries );
+		}
+	}
+
+	/**
+	 * Adds an entry to the MO structure and caches it.
+	 *
+	 * @since 3.7
+	 *
+	 * @param Translation_Entry|array $entry The entry to add.
+	 * @return bool True if the entry was added, false otherwise.
+	 */
+	public function add_entry( $entry ) {
+		if ( is_array( $entry ) ) {
+			$entry = new Translation_Entry( $entry );
+		}
+
+		if ( parent::add_entry( $entry ) ) {
+			$language = $this->get_header( 'Language' );
+			if ( ! empty( $language ) ) {
+				self::$cache->set( $language, $this->entries );
+			}
+
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Adds or merges an entry to the MO structure and caches it.
+	 *
+	 * @since 3.7
+	 *
+	 * @param Translation_Entry|array $entry The entry to add.
+	 * @return bool True if the entry was added, false otherwise.
+	 */
+	public function add_entry_or_merge( $entry ) {
+		if ( is_array( $entry ) ) {
+			$entry = new Translation_Entry( $entry );
+		}
+
+		if ( parent::add_entry_or_merge( $entry ) ) {
+			$language = $this->get_header( 'Language' );
+			if ( ! empty( $language ) ) {
+				self::$cache->set( $language, $this->entries );
+			}
+
+			return true;
+		}
+
+		return false;
```

Comment:
Isn't this overkill? Where do we need to reset the cache for these methods in addition to the reset already done when saving the translations?

Reply:
I kinda disagree, even though in the code it seems `export_to_db()` is always called after `add_entry()`, `PLL_MO` should be used without `export_to_db()` for it to behave correctly IMHO.
I proposed a simplification to `add_entry()` and `add_entry_or_merge()` in e8532418.
That said, maybe populating the cache in `import_from_db()` is useless, do you want me to remove that instead?

Reply:
I still believe that we should not clear the cache in `delete_entry()`, `add_entry()` and `add_entry_or_merge()`.
Manipulating entries in a a first object would impact the cache of a second object without benefit.

Reply:
Removed in a6645c1e.